lib/common/asar.js

```diff
@@ -373,6 +408,18 @@
       })
     }
 
+    fs.exists[util.promisify.custom] = function (p) {
+      const [isAsar, asarPath, filePath] = splitPath(p)
+      if (!isAsar) {
+        return exists[util.promisify.custom](p)
+      }
+      const archive = getOrCreateArchive(asarPath)
+      if (!archive) {
+        return Promise.resolve(false)
```

Comment:
This is inconsistent with the `fs.exists` behavior above, in the case of an invalid archive it throws / calls back with an error.  This method will fail transparently with a resolved `false`

Reply:
@MarshallOfSound original `fs.exists` never throws, it always returns either `true` or `false`. I can make it reject in  this case though.